### PR TITLE
IncomesKeeper shouldn't depend on PaymentProcessor

### PR DESCRIPTION
**Pull request by Krigpl**
Because it just doesn't make sense to make outgoing payments through a class that is responsible for incomes.
IncomesKeeper only needs an access to the token, not the whole PaymentProcessor.

### CODE REVIEW

**Review by etam — APPROVED** [review]

**Review by etam — CHANGES_REQUESTED** [review]
Tests are failing. Pylint is sad.

**Comment by Krigpl** [comment]
Pylint will remain sad because I can't do much about those errors
 ``` E: tests/golem/transactions/ethereum/test_ethereumtransactionsystem.py:100:19: Instance of 'PaymentProcessor' has no '_PaymentProcessor__client' member (no-member)```
(or can I)?

**Comment by Krigpl** [comment]
I guess we could update pylint https://github.com/PyCQA/pylint/issues/1643

**Comment by etam** [comment]
Use `_client` instead of `__client`?

**Comment by etam** [comment]
Updating pylint is also a good idea.

**Comment by Krigpl** [comment]
> Use _client instead of __client?

I'm aiming to remove `Client` from `PaymentProcessor` altogether so this will resolve itself anyway.

File: golem/transactions/ethereum/ethereumincomeskeeper.py [hunk 1]
```diff
@@ -1,29 +1,20 @@
...
-    def __init__(self, processor: PaymentProcessor) -> None:
-        self.processor = processor
-
-    def start(self):
-        self.processor.start()
-
-    def stop(self):
-        if self.processor.running:
-            self.processor.stop()
+    def __init__(self, eth_address: str, token) -> None:
```
Comment (mfranciszkiewicz) [inline golem/transactions/ethereum/ethereumincomeskeeper.py:15]: Just a question - why not `client` instead of `token`?

Comment (Krigpl) [reply golem/transactions/ethereum/ethereumincomeskeeper.py:15]: Because of `get_incomes_from_block` which is token dependent.

**Review by mfranciszkiewicz — APPROVED** [review]

**Review by etam — APPROVED** [review]

**Pull request merged**